### PR TITLE
Nested routes for organization/id/members and auth for organizations

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

#145 
#113 

## Purpose/Implementation Notes

I mostly followed the patterns from the grants route.

@dvenprasad I decided that only users who are already members of an organization can be promoted to owner, so that way people can't spam by creating orgs and giving them away to unwilling users.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

Leanin on the unit tests pretty hard.

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/resources_portal/test/views/test_organization.py [hunk 1]
```diff
@@ -49,46 +55,83 @@ def setUp(self):
...
         organization_json = self.client.get(self.url).json()
 
         new_member = UserFactory()
         new_member_json = {"id": new_member.id}
-        # The lab gets sold to a new owner or something...
+        # Someone is trying to steal the lab!!!
         organization_json["owner"] = new_member_json
         organization_json["members"].append(new_member_json)
 
         # TODO: this should require authentication
```
Comment (Philinphiladelphia) [inline api/resources_portal/test/views/test_organization.py:78]: this TODO has been done

File: api/resources_portal/test/views/test_organization.py [hunk 2]
```diff
@@ -49,46 +55,83 @@ def setUp(self):
...
+        self.client.force_authenticate(user=user)
         response = self.client.get(self.url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
-    def test_put_request_updates_a_organization(self):
+    def test_get_requires_account(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 403)
+
+    def test_put_request_fails_by_stranger(self):
```
Comment (Philinphiladelphia) [inline api/resources_portal/test/views/test_organization.py:67]: The test name is a little unclear. Is this testing that a PUT request by the wrong person fails?

File: api/resources_portal/test/views/test_organization.py [hunk 3]
```diff
@@ -49,46 +55,83 @@ def setUp(self):
...
-        new_member = User.objects.get(id=new_member.id)
         self.organization.refresh_from_db()
-        self.assertIn(new_member, self.organization.members.all())
-        self.assertEqual(new_member, self.organization.owner)
+        # The owner always gets added to the members list:
+        self.assertIn(new_owner, self.organization.members.all())
+        self.assertEqual(new_owner, self.organization.owner)
+        self.assertEqual(new_name, self.organization.name)
+
+        # But adding members requries a request to the nested route:
```
Comment (Philinphiladelphia) [inline api/resources_portal/test/views/test_organization.py:113]:
```suggestion
        # But adding members requires a request to the nested route:
```

File: api/resources_portal/test/views/test_organization.py [hunk 4]
```diff
@@ -49,46 +55,83 @@ def setUp(self):
...
+        self.assertNotIn(new_member, self.organization.members.all())
+
+    def test_put_owner_fails_if_not_member(self):
+        """The new owner must belong to the organization already."""
+        self.client.force_authenticate(user=self.organization.owner)
+        organization_json = self.client.get(self.url).json()
+
+        new_owner = UserFactory()
+        new_owner_json = {"id": new_owner.id}
+        # Ahhh! A user is trying to steal the lab!!!
```
Comment (Philinphiladelphia) [inline api/resources_portal/test/views/test_organization.py:123]: Lmao these make the PR a lot more exciting to read. It's like a drama

File: api/resources_portal/test/views/test_organization.py [hunk 5]
```diff
@@ -99,3 +142,9 @@ def test_put_cannot_add_invalid_user(self):
         # TODO: this should require authentication
```
Comment (Philinphiladelphia) [inline api/resources_portal/test/views/test_organization.py:142]: This TODO has been done, right?

File: api/resources_portal/test/views/test_organization.py [hunk 6]
```diff
@@ -49,46 +55,83 @@ def setUp(self):
         self.url = reverse("organization-detail", args=[self.organization.id])
 
     def test_get_request_returns_a_given_organization(self):
```
Comment (Philinphiladelphia) [inline api/resources_portal/test/views/test_organization.py:57]: Do we want a test case for unauthenticated (anonymous) users failing? To login as anonymous user, use `self.client.force_authenticate(user=None)`

Comment (kurtwheeler) [reply api/resources_portal/test/views/test_organization.py:57]: How is this different than just not calling `self.client.force_authenticate` at all? Aren't the requests by default made by a unauthenticated user?

Comment (Philinphiladelphia) [reply api/resources_portal/test/views/test_organization.py:57]: That's probably true. Still, might be good to have an unauthenticated test case.

**Comment by kurtwheeler** [comment]
@Philinphiladelphia I believe I've addressed your feedback.

File: api/resources_portal/test/views/test_organization.py (hunk 6 above)
Comment (kurtwheeler) [reply api/resources_portal/test/views/test_organization.py:57]: Done.

**Pull request merged**